pkg/kn/commands/service/configuration_edit_flags.go

```diff
@@ -51,10 +51,10 @@ func (p *ConfigurationEditFlags) AddUpdateFlags(command *cobra.Command) {
 	command.Flags().StringVar(&p.RequestsFlags.Memory, "requests-memory", "", "The requested memory (e.g., 64Mi).")
 	command.Flags().StringVar(&p.LimitsFlags.CPU, "limits-cpu", "", "The limits on the requested CPU (e.g., 1000m).")
 	command.Flags().StringVar(&p.LimitsFlags.Memory, "limits-memory", "", "The limits on the requested memory (e.g., 1024Mi).")
-	command.Flags().IntVar(&p.MinScale, "min-scale", 0, "Minimal number of replicas.")
-	command.Flags().IntVar(&p.MaxScale, "max-scale", 0, "Maximal number of replicas.")
+	command.Flags().IntVar(&p.MinScale, "min-scale", -1, "Minimal number of replicas.")
```

Comment:
`-1` reads odd since it's kind of an impossible number. What does -1 minimum replica means... same for the others.

Reply:
I intentionally put the invalid number on these update options as default. Let's say, someone wants to change the minScale value for hello service from 1 to 0 and runs `kn service update hello --min-scale 0`. In the previous implementation, kn didn't allow to set 0 value to minScale option according to the validation below. Therefore, the users cannot update minScale value as they want. I think this behavior is not desirable.

https://github.com/knative/client/blob/144ccb875e1de81067623281e5628e67d2c982a4/pkg/serving/config_changes.go#L40-L42

That's why I set the invalid number to those update options as default. If we need the default values for each option, I guess we should put the invalid numbers on them and change the value only when the users pass the valid number for each option. Otherwise, leave it as it is.

Reply:
I think it's better to use https://godoc.org/github.com/spf13/pflag#FlagSet.Changed, which will tell you straight-up whether the user set it. That way things that read the defaults don't have to worry about weird invalid numbers.

Reply:
@sixolet Thanks for your suggestion! I updated to use FlagSet.Changed so we don't care about default value for those update options.